packages/x-charts/src/ChartsLabel/ChartsLabelMark.tsx

```diff
@@ -0,0 +1,199 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { styled, SxProps, Theme } from '@mui/material/styles';
+import clsx from 'clsx';
+import { ChartsLabelMarkClasses, labelMarkClasses, useUtilityClasses } from './labelMarkClasses';
+import { consumeThemeProps } from '../internals/consumeThemeProps';
+
+export interface ChartsLabelMarkProps {
+  /**
+   * Defines the max size of the mark.
+   *
+   * For the `line` type, the size is the length of the line.
+   * For all other types, the size is the width and height of the mark.
+   *
+   * @default type='square': 13
+   * @default type='line': 16
+   * @default type='circle': 15
+   */
+  size?: number;
+  /**
+   * The type of the mark.
+   * @default 'square'
+   */
+  type?: 'square' | 'circle' | 'line' | (string & {});
```

Comment:
More a follow-up than a comment for this PR. You could either

- use a generic for this typing and make it consistent with the way custom SVG are manage at the charts level
- extend it's type with the new supported types are added by using the theme provider